### PR TITLE
BREAKING(flags): introduce negatable option

### LINKED ISSUE #2300: Flags: Make negatable options configurable
**Is your feature request related to a problem? Please describe.**

We discussed in #2180 that we want to make negatable options configurable, because it would improve the typings we want to introduce in #2180.

**Describe the solution you'd like**

We agreed for now on following solution:

```ts
parse(Deno.args, {
  string: ["config"],
  negatable: ["config"],
});
```

**Describe alternatives you've considered**

An alternative solution could be to prefix the option name with `no-`.

```ts
parse(Deno.args, {
  string: ["config", "no-config"],
});
```

### PR DESCRIPTION
**Pull request by c4spar**
close #2300

### CODE REVIEW

File: flags/test.ts [hunk 1]
```diff
@@ -197,37 +197,57 @@ Deno.test("booleanParsingTrueLike", function (): void {
...
+  const parsed2 = parse(["--foo", "--no-foo", "123"], {
+    boolean: ["foo"],
+    negatable: ["foo"],
+  });
   assertEquals(parsed2.foo, false);
 });
 
 Deno.test("booleanAfterBooleanNegation", function (): void {
-  const parsed = parse(["--no--foo", "--foo"], { boolean: ["foo"] });
+  const parsed = parse(["--no--foo", "--foo"], {
```
Comment (kt3k) [inline flags/test.ts:214]: nit: Not an error of this change, but I think this `"--no--foo"` should be `"--no-foo"`.

`parsed` here looks like `{ _: [], "no--foo": true, foo: true }`, and that doesn't check what the test name suggests.

**Review by kt3k — COMMENTED** [review]
Mostly looks good. Nice work!

Left one comment

File: flags/mod.ts [hunk 2]
```diff
@@ -201,6 +208,22 @@ export function parse(
     }
   }
 
+  if (negatable !== undefined) {
+    const negatableArgs = typeof negatable === "string"
```
Comment (cjihrig) [inline flags/mod.ts:212]: Does it make sense to perform some input validation here? What if `negatable` is not a string, but also not an array? Or, what if it is an array that includes things besides strings. Or, what if the provided values are strings, but nonsense values? It feels like we should either have validation, tests, or both for these edge cases.

File: flags/test.ts (hunk 1 above)
Comment (c4spar) [reply flags/test.ts]: fixed

**Review by kt3k — APPROVED** [review]
@c4spar Thanks for updating the test cases. LGTM.

**Comment by kt3k** [comment]
Note: This is a breaking change when the user's CLI tool accepting `--no-xyz`-form flags. In that case the user's CLI tool needs to specify that arg name in `negatable` option (So the workaround is obvious and doesn't require lot of changes to their code)

The main motivation of introducing this breaking change is to make the typing (in #2180) more useful and reasonable. Without this change, all args can be `false` even when the arg is specified as `string`. This PR changes it, and the API now accepts `--no-xyz` as negation of `--xyz` only when `xyz` is specified in `negatable` option.
